src/js/pages/pipeline/Pipeline.jsx

```diff
@@ -0,0 +1,175 @@
+import React, { useState, useEffect } from 'react';
+
+import PipelineContext from 'js/context/Pipeline';
+import { useAuth0 } from 'js/context/Auth0';
+import { useUserContext } from 'js/context/User';
+import { useFiltersContext } from 'js/context/Filters';
+
+import { getMetrics, fetchApi } from 'js/services/api';
+import { number } from 'js/services/format';
+
+import { palette } from 'js/res/palette';
+
+const distinct = (collection, extractor) => {
+    return Object.keys(
+        collection.reduce((acc, item) => {
+            const processed = extractor(item);
+            if (Array.isArray(processed)) {
+                processed.forEach(element => acc[element] = true);
+            } else {
+                acc[processed] = true;
+            }
+
+            return acc;
+        }, {})
+    );
+};
```

Comment:
What about using `Set` and `flatMap`?

```suggestion
const distinct = (collection, extractor) => (new Set(x.flatMap((v) => extractor(v)))).keys();
```

Reply:
what a shame...
I should study the methods of `Array.prototype` :(